frontend/src/components/BarChartWidget.tsx

```diff
@@ -151,122 +91,32 @@ const BarChartWidget = (props: Props) => {
-                        <BarChart
-                            className="bar-chart"
-                            data={props.data}
-                            layout="vertical"
-                            margin={{ right: 0, left: 0 }}
-                            ref={(el: CategoricalChartWrapper) => {
-                                chartRef.current = el;
-                                setChartLoaded(!!el);
-                            }}
-                        >
-                            <CartesianGrid horizontal={false} />
-                            <XAxis
-                                type="number"
-                                tickFormatter={(tick: any) =>
-                                    TickFormatter.format(
-                                        Number(tick),
-                                        xAxisLargestValue,
-                                        props.significantDigitLabels,
-                                        "",
-                                        "",
-                                    )
-                                }
-                            />
-                            <YAxis
-                                dataKey={props.bars.length ? props.bars[0] : ""}
-                                type={yAxisType()}
-                                width={dims.labelWidth}
-                                minTickGap={0}
-                                domain={["dataMin - 1", "dataMax + 1"]}
-                                scale={yAxisType() === "number" ? "linear" : "auto"}
-                                tickFormatter={formatYAxisLabel}
-                                reversed={true}
-                            />
-                            <Tooltip
-                                itemStyle={{ color: "#1b1b1b" }}
-                                isAnimationActive={false}
-                                formatter={(value: number | string, name: string) => {
-                                    // Check if there is metadata for this column
-                                    let columnMetadata;
-                                    if (props.columnsMetadata) {
-                                        columnMetadata = props.columnsMetadata.find(
-                                            (cm) => cm.columnName === name,
-                                        );
-                                    }
-
-                                    return TickFormatter.format(
-                                        Number(value),
-                                        xAxisLargestValue,
-                                        props.significantDigitLabels,
-                                        "",
-                                        "",
-                                        columnMetadata,
-                                    );
-                                }}
-                            />
-                            {!props.hideLegend && (
```

Comment:
hideLegend is also missing

Reply:
check showMobilePreview, In general, we don't want to break any existing contract outside of the component. Please, go over each property of the props and ensure they are used in the new chart.

Reply:
`hideLegend` is a legacy property not linked to the UI. it's being removed from the component contract.